Description: Add support for building python3 wheel in Debian Openstack
Author: Chuck Short <charles.short@windriver.com>m>
---
 build-tools/pkgos-dh_auto_install | 9 +++++++++
 debian/changelog                  | 6 ++++++
 debian/control                    | 1 +
 3 files changed, 16 insertions(+)

diff --git a/build-tools/pkgos-dh_auto_install b/build-tools/pkgos-dh_auto_install
index f766f13..dcccb5b 100755
--- a/build-tools/pkgos-dh_auto_install
+++ b/build-tools/pkgos-dh_auto_install
@@ -18,6 +18,10 @@ for i in $@ ; do
 		PKGOS_IN_TMP=yes
 		shift
 		;;
+	"--wheel")
+		PKGOS_USE_WHEEL=yes
+		shift
+		;;
 	*)
 		;;
 	esac
@@ -36,5 +40,10 @@ fi
 for pyvers in ${PYTHON3S}; do
 	python${pyvers} setup.py install --install-layout=deb --root $(pwd)/debian/${TARGET_DIR}
 done
+	if [ "${PKGOS_USE_WHEEL}" = "yes" ]; then
+		for pyvers in ${PYTHON3S}; do
+			python${pyvers} setup.py bdist_wheel --universal  -d $(pwd)/debian/python3-${PY_MODULE_NAME}-wheel/usr/share/python-wheel
+		done
+	fi
 rm -rf $(pwd)/debian/python*/usr/lib/python*/dist-packages/*.pth
 rm -rf $(pwd)/debian/tmp/usr/lib/python*/dist-packages/*.pth
diff --git a/debian/changelog b/debian/changelog
index 63aa502..4533066 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -91,6 +91,12 @@ openstack-pkg-tools (118) unstable; urgency=medium
   * Add code to calculate swift-object multi-port uwsgi startup.
 
  -- Thomas Goirand <zigo@debian.org>  Mon, 18 Jan 2021 21:41:08 +0100
+openstack-pkg-tools (117+nmu1) unstable; urgency=medium
+
+  * Non-maintainer upload.
+  * build-tools/pkgos-dh_auto_install: Add wheel support.
+
+ -- Chuck Short <zulcss@ubuntu.com>  Sun, 03 Oct 2021 15:10:16 +0000
 
 openstack-pkg-tools (117) unstable; urgency=medium
 
diff --git a/debian/control b/debian/control
index 8cb1b2d..70ed450 100644
--- a/debian/control
+++ b/debian/control
@@ -17,6 +17,7 @@ Architecture: all
 Multi-Arch: foreign
 Depends:
  python3-pip,
+ python3-wheel,
  gettext,
  jq,
  po-debconf,
-- 
2.34.1

